ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/pull-queries-with-predicates.json

```diff
@@ -0,0 +1,89 @@
+{
+  "comments": [
+    "Tests covering pull queries with predicates"
+  ],
+  "tests": [
+    {
+      "name": "like operator on key and non-key columns",
+      "statements": [
+        "CREATE TABLE INPUT (ID STRING PRIMARY KEY, V0 STRING, V1 STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE TABLE MATVIEW AS SELECT ID, V0, V1 FROM INPUT;",
+        "SELECT ID, V0, V1 FROM MATVIEW WHERE ID LIKE '%1';",
+        "SELECT ID, V0, V1 FROM MATVIEW WHERE V0 LIKE 'v12%';"
+      ],
+      "properties": {
+        "ksql.query.pull.table.scan.enabled":  true
+      },
+      "inputs": [
+        {"topic": "test_topic", "timestamp": 12345, "key": "10", "value": {"v0": "v10a", "v1": "v10b"}},
+        {"topic": "test_topic", "timestamp": 12355, "key": "11", "value": {"v0": "v11a", "v1": "v11b"}},
+        {"topic": "test_topic", "timestamp": 12365, "key": "12", "value": {"v0": "v12a", "v1": "v12b"}},
+        {"topic": "test_topic", "timestamp": 12375, "key": "13", "value": {"v0": "v13a", "v1": "v13b"}},
+        {"topic": "test_topic", "timestamp": 12385, "key": "14", "value": {"v0": "v14a", "v1": "v14b"}}
+      ],
+      "responses": [
+        {"admin": {"@type": "currentStatus"}},
+        {"admin": {"@type": "currentStatus"}},
+        {"query": [
+          {"header":{"schema":"`ID` STRING KEY, `V0` STRING, `V1` STRING"}},
+          {"row":{"columns":["11", "v11a", "v11b"]}}
+        ]},
+        {"query": [
+          {"header":{"schema":"`ID` STRING KEY, `V0` STRING, `V1` STRING"}},
+          {"row":{"columns":["12", "v12a", "v12b"]}}
+        ]}
+      ]
+    },
+    {
+      "name": "like operator must fail on non-varchar columns",
+      "statements": [
+        "CREATE TABLE INPUT (ID INTEGER PRIMARY KEY, V0 STRING, V1 STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE TABLE MATVIEW AS SELECT ID, V0, V1 FROM INPUT;",
+        "SELECT ID, V0, V1 FROM MATVIEW WHERE ID LIKE '%12';"
+      ],
+      "properties": {
+        "ksql.query.pull.table.scan.enabled":  false
+      },
+      "expectedError": {
+        "message": "The column type for Like condition must be VARCHAR. The column type is INTEGER"
+      }
+    },
+    {
+      "name": "between operator on key and non-key columns",
+      "statements": [
+        "CREATE STREAM INPUT (ID1 STRING KEY, ID2 INT, V0 STRING) WITH (kafka_topic='test_topic', format='JSON');",
+        "CREATE TABLE AGGREGATE AS SELECT ID1, ID2, V0, COUNT(1) AS COUNT FROM INPUT GROUP BY ID1, ID2, V0;",
+        "SELECT * FROM AGGREGATE WHERE ID1 BETWEEN '10' and '11';",
```

Comment:
Might want some error cases where ID1 isn't a column and its type disagrees with the literals.

Reply:
More tests are added.